File: C언어/SWEA/4613/SWEA_4613.c
#include <stdio.h>

int main(void) {
	int T, N, M, data[50][3];
	char arr[50][50];

	scanf("%d", &T);

	for (int tc = 1; tc <= T; tc++) {
		scanf("%d %d", &N, &M);

		for (int i = 0; i < N; i++) {
			int W = 0, B = 0, R = 0;
			scanf("%s", arr[i]);
			for (int j = 0; j < M; j++) {
				if (arr[i][j] == 'W')
					W += 1;
				else if (arr[i][j] == 'B')
					B += 1;
				else
					R += 1;
			}
			data[i][0] = W;
			data[i][1] = B;
			data[i][2] = R;
		}

		int max_res = 0;
		for (int white = 0; white < N - 2; white++) {
			for (int blue = white + 1; blue < N - 1; blue++) {
				int res = 0;
				for (int i = 0; i <= white; i++) {
					res += data[i][0];
				}
				for (int i = white + 1; i <= blue; i++) {
					res += data[i][1];
				}
				for (int i = blue + 1; i < N; i++) {
					res += data[i][2];
				}

				if (max_res < res)
					max_res = res;
			}
		}
		printf("#%d %d\n", tc, N * M - max_res);
	}
	return 0;
}

File: SWEA_4613.py
# 2020.03.11
# 18:59 ~ 19:19
# 전탐
# 시간:167ms, 코드 길이:853B

for tc in range(1, int(input())+1):
    N, M = map(int,input().split())
    arr = [list(input()) for _ in range(N)]
    ans = 0
    line_data = []

    for i in range(N):
        W, B, R = 0, 0, 0

        for j in range(M):
            if arr[i][j] == "W":
                W += 1
            elif arr[i][j] == "B":
                B += 1
            else:
                R += 1
        line_data.append((W, B, R))

    max_res = 0
    for white in range(N-2):
        for blue in range(white+1, N-1):
            res = 0
            for i in range(white+1):
                res += line_data[i][0]
            for i in range(white+1, blue+1):
                res += line_data[i][1]
            for i in range(blue+1, N):
                res += line_data[i][2]

            max_res = max(max_res, res)

    print("#%d"%(tc), N*M-max_res)
